src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4803,6 +4805,10 @@ private void ParseVariableDeclarators(TypeSyntax type, VariableFlags flags, Sepa
                 {
                     break;
                 }
+                else if (stopOnCloseParen && this.CurrentToken.Kind == SyntaxKind.CloseParenToken)
+                {
+                    break;
+                }
```

Comment:
this is half of the fix.  Callers who are parsing a list of declarators say how close parens should be treated.